compiler/rustc_interface/src/passes.rs

```diff
@@ -757,14 +757,28 @@ fn analysis(tcx: TyCtxt<'_>, (): ()) -> Result<()> {
                 CStore::from_tcx(tcx).report_unused_deps(tcx);
             },
             {
+                // Prefetch this as it is used later by the loop below
+                // to prevent multiple threads from blocking on it.
+                let _ = tcx.get_lang_items(());
```

Comment:
Can this be fixed in query infra, yielding control to another job if blocked on another query?

Reply:
Somewhat by using a thread pool with fibers. That's not a short term fix and it's not free, so we might want to keep these anyway.